File: 2026-01/src/day_15/boj_1485/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int t;
    cin >> t;

    for (int tc = 1; tc <= t; tc++) {
        vector<pair<long long, long long>> v(4);
        for (auto& p : v) cin >> p.first >> p.second;
        sort(v.begin(), v.end());

        long long side1 = (v[0].first - v[1].first) * (v[0].first - v[1].first) +
                          (v[0].second - v[1].second) * (v[0].second - v[1].second);
        long long side2 = (v[0].first - v[2].first) * (v[0].first - v[2].first) +
                          (v[0].second - v[2].second) * (v[0].second - v[2].second);
        long long side3 = (v[3].first - v[1].first) * (v[3].first - v[1].first) +
                          (v[3].second - v[1].second) * (v[3].second - v[1].second);
        long long side4 = (v[3].first - v[2].first) * (v[3].first - v[2].first) +
                          (v[3].second - v[2].second) * (v[3].second - v[2].second);
        long long diagonal1 = (v[3].first - v[0].first) * (v[3].first - v[0].first) +
                              (v[3].second - v[0].second) * (v[3].second - v[0].second);
        long long diagonal2 = (v[2].first - v[1].first) * (v[2].first - v[1].first) +
                              (v[2].second - v[1].second) * (v[2].second - v[1].second);

        cout << (side1 == side2 && side1 == side3 && side1 == side4 && diagonal1 == diagonal2) << '\n';
    }
}

File: 2026-01/src/day_15/boj_1485/Main.java
package day_15.boj_1485;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;

        int T = Integer.parseInt(br.readLine());
        for (int tc = 1; tc <= T; tc++) {
            long[][] pos = new long[4][2];
            for (int i = 0; i < 4; i++) {
                st = new StringTokenizer(br.readLine());
                pos[i][0] = Long.parseLong(st.nextToken());
                pos[i][1] = Long.parseLong(st.nextToken());
            }
            Arrays.sort(pos, (o1, o2) -> {
                if (o1[0] != o2[0]) return Long.compare(o1[0], o2[0]);
                return Long.compare(o1[1], o2[1]);
            });

            long side1 = (pos[0][0] - pos[1][0]) * (pos[0][0] - pos[1][0]) + (pos[0][1] - pos[1][1]) * (pos[0][1] - pos[1][1]);
            long side2 = (pos[0][0] - pos[2][0]) * (pos[0][0] - pos[2][0]) + (pos[0][1] - pos[2][1]) * (pos[0][1] - pos[2][1]);
            long side3 = (pos[3][0] - pos[1][0]) * (pos[3][0] - pos[1][0]) + (pos[3][1] - pos[1][1]) * (pos[3][1] - pos[1][1]);
            long side4 = (pos[3][0] - pos[2][0]) * (pos[3][0] - pos[2][0]) + (pos[3][1] - pos[2][1]) * (pos[3][1] - pos[2][1]);
            long diagonal1 = (pos[3][0] - pos[0][0]) * (pos[3][0] - pos[0][0]) + (pos[3][1] - pos[0][1]) * (pos[3][1] - pos[0][1]);
            long diagonal2 = (pos[2][0] - pos[1][0]) * (pos[2][0] - pos[1][0]) + (pos[2][1] - pos[1][1]) * (pos[2][1] - pos[1][1]);

            if (side1 == side2 && side1 == side3 && side1 == side4 && diagonal1 == diagonal2) {
                sb.append(1).append("\n");
            } else {
                sb.append(0).append("\n");
            }
        }

        System.out.print(sb);
    }
}
